troubleshooting/incrementally_configuring_images.md

```diff
@@ -0,0 +1,297 @@
+
+   ```bash
+   export CFS_CONFIGURATION_NAME="ssi-compute-23.11.0-SSI-csm-1.5.0.beta37-10"
+   ```
+
+   SAT may have generated multiple configurations, in which case it is up to the user to determine which is the desired configuration.  However if the default SAT `bootprep` file is being used, the configuration name should include `compute`
+
+1. (`ncn-mw#`) Set `SESSION_NAME`.
+
+   This can be set to any value, but the following example steps assume this variable is set.
+
+   ```bash
+   export SESSION_NAME="example-partial-image-configuration"
+   ```
+
+1. (Optional) Find the failed configuration layer.
+
+   If configuration of the image failed somewhere beyond the first configuration layer, and there is some confidence that the successful layers will continue to be successful,
+   it is possible to save time by generating the first partial image with all of the successful layers.
+   Find the index of the of the failed configuration either by counting the number of successful configuration layers in the logs, or by comparing the name of the failed layer to the output of `cray cfs v3 configurations describe $CFS_CONFIGURATION_NAME`.
+   Indices for the configuration layers start at 0.
+
+## Generating a partially configured image
+
+1. (`ncn-mw#`) Set the configuration limit.
+
+   If this is the first CFS run of this procedure, this should be `0` to apply only the first layer.
+
+   ```bash
+   export CONFIGURATION_LIMIT=0
+   ```
+
+   Optionally, this can instead be set to a comma-separated list of numbers starting at 0 when trying to apply all layers up to the failed layer. e.g.
+
+      ```bash
+   export CONFIGURATION_LIMIT=0,1,2
+   ```
+
+   If this is not the first CFS run in this procedure, set this value to the next layer to be applied.
+
+   ```bash
+   export CONFIGURATION_LIMIT=<previous CONFIGURATION limit +1>
+   ```
+
+1. (`ncn-mw#`) Generate a partial configured image.
+
+    The following command will generate an image that only adds the configuration layers specified in the `CONFIGURATION_LIMIT`.
+
+   ```bash
+   cray cfs v3 sessions create --name $SESSION_NAME --configuration-name $CFS_CONFIGURATION_NAME --target-definition image --target-group Compute $BASE_IMAGE_ID --configuration-limit $CONFIGURATION_LIMIT
+   ```
+
+   If this is the last layer of the configuration, it is optionally possible to specify the name of the resulting image by adding `--target-image-map`.
+
+   ```bash
+   cray cfs v3 sessions create --name $SESSION_NAME --configuration-name $CFS_CONFIGURATION_NAME --target-definition image --target-group Compute $BASE_IMAGE_ID --configuration-limit $CONFIGURATION_LIMIT --target-image-map $BASE_IMAGE_ID <desired image name>
+   ```
+
+1. (`ncn-mw#`) Monitor CFS and retrieve the partial image id.
+
+    Monitor the CFS session with the following command until the status is complete.  If `suceeded` is true, move on to the next step, otherwise debug the failure and re-run the session as necessary.
```

Comment:
```suggestion
    Monitor the CFS session with the following command until the status is complete. If `suceeded` is true, then move on to the next step; otherwise, debug the failure and re-run the session as necessary.
```